state/sqlserver/migration.go

```diff
@@ -273,92 +299,88 @@ func (m *migration) createStoredProcedureIfNotExists(db *sql.DB, name string, es
 		name,
 		escapedDefinition)
 
-	return runCommand(tsql, db)
+	return runCommand(ctx, db, tsql)
 }
 
 /* #nosec. */
 //nolint:dupword
-func (m *migration) ensureUpsertStoredProcedureExists(db *sql.DB, mr migrationResult) error {
+func (m *migration) ensureUpsertStoredProcedureExists(ctx context.Context, db *sql.DB, mr migrationResult) error {
 	tsql := fmt.Sprintf(`
-			CREATE PROCEDURE %s (
-				@Key 			%s,
+			CREATE PROCEDURE %[1]s (
+				@Key 			%[2]s,
 				@Data 			NVARCHAR(MAX),
-				@RowVersion		BINARY(8),
-				@FirstWrite		BIT)
-			AS
+				@TTL INT,
+				@RowVersion	BINARY(8),
+				@FirstWrite	BIT
+			) AS
+			BEGIN
 				IF (@FirstWrite=1)
 					BEGIN
 						IF (@RowVersion IS NOT NULL)
 							BEGIN
 								BEGIN TRANSACTION;
-								IF NOT EXISTS (SELECT * FROM [%s] WHERE [KEY]=@KEY AND RowVersion = @RowVersion)
+								IF NOT EXISTS (SELECT * FROM [%[3]s] WHERE [KEY]=@KEY AND RowVersion = @RowVersion)
 									BEGIN
 										THROW 2601, ''FIRST-WRITE: COMPETING RECORD ALREADY WRITTEN.'', 1
 									END
 								BEGIN
-									UPDATE [%s]
-									SET [Data]=@Data, UpdateDate=GETDATE()
+									UPDATE [%[3]s]
+									SET [Data]=@Data, UpdateDate=CURRENT_TIMESTAMP, ExpireDate=CASE WHEN @TTL IS NULL THEN NULL ELSE DATEADD(SECOND, @TTL, CURRENT_TIMESTAMP) END
```

Comment:
I think ExpireDate needs to be updated regardless, not just if null?

Reply:
Yes, `ExpireDate` is being `SET` to `NULL` when `@TTL` is `NULL`.

Reply:
Ok thanks for confirming, I did not understand the T-SQL :)